### PR TITLE
Make waste heat inputs optional for Coil:*:DX:Multispeed

### PR DESCRIPTION
**Pull request by lgu1234**
Addresses #5162 
For Coil:Cooling:DX:MultiSpeed and Coil:Heating:DX:MultiSpeed the followng fields are no longer required.
Speed n Rated Waste Heat Fraction of Power Input now defaults to 0.2.
Speed n Waste Heat Function of Temperature Curve Name will default to a constant of 1.0 if left blank.

### CODE REVIEW

**Comment by mjwitte** [comment]
@lgu1234 Merging #5144 created conflicts here.

**Comment by mjwitte** [comment]
@lgu1234 Do you expect this one to be for v8.4.0 or v8.5.0?

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -4095,13 +4095,17 @@ namespace DXCoils {
...
+							ShowWarningError( RoutineName + CurrentModuleObject + "=\"" + DXCoil( DXCoilNum ).Name + "\", missing" );
+							ShowContinueError( "... " + cAlphaFields( 18 + ( I - 1 ) * 6 ) + " is blank." );
 						} else {
-							ShowSevereError( RoutineName + CurrentModuleObject + "=\"" + DXCoil( DXCoilNum ).Name + "\", invalid" );
+							ShowWarningError( RoutineName + CurrentModuleObject + "=\"" + DXCoil( DXCoilNum ).Name + "\", invalid" );
 							ShowContinueError( "...not found " + cAlphaFields( 18 + ( I - 1 ) * 6 ) + "=\"" + Alphas( 18 + ( I - 1 ) * 6 ) + "\"." );
 						}
+<<<<<<< .mine
+//						ErrorsFound = true;
+=======
```
Comment (mjwitte) [inline src/EnergyPlus/DXCoils.cc]: @lgu1234 The builds are failing due to these conflict markers <<<< .mine and below >>>> .theirs.  Also, the end of the unit test file needs a newline.

**Comment by mjwitte** [comment]
@lgu1234 I was expecting both the waste heat fraction and curve name to be optional. The fraction is still required.  And I see warnings if the fuel type is naturalgas with a blank curve type.  All four of the 5162 test files should run without any warnings.

**Comment by lgu1234** [comment]
@mjwitte Here is a logic I follow. The GetDXCoils is called first by the parent object to read coil inputs. 

```
                if ( DXCoil( DXCoilNum ).MSWasteHeat( I ) == 0 ) {
                    if ( lAlphaBlanks( 18 + ( I - 1 ) * 6 ) ) {
                        ShowWarningError( RoutineName + CurrentModuleObject + "=\"" + DXCoil( DXCoilNum ).Name + "\", missing" );
                        ShowContinueError( "... " + cAlphaFields( 18 + ( I - 1 ) * 6 ) + " is blank." );
                    } else {
                        ShowWarningError( RoutineName + CurrentModuleObject + "=\"" + DXCoil( DXCoilNum ).Name + "\", invalid" );
                        ShowContinueError( "...not found " + cAlphaFields( 18 + ( I - 1 ) * 6 ) + "=\"" + Alphas( 18 + ( I - 1 ) * 6 ) + "\"." );
                    }
                } else {
```

At that time, coils don't know whether the parent object requires waste heat recovery or not. The above check provides warnings to show the input is either blank or invalid. The heat recovery check was accomplished in InitDXCoil, where I lose information on either blank input or invalid input. If I remove these warnings in GetDXCoil, a user may not know what is wrong about input: blank or invalid. That is why I intend to keep these warnings. An alternative choice is to use DisplayExtraWarning to cover it, so that no warnings will be shown in the test files.    

What do you think?

**Comment by mjwitte** [comment]
For the waste heat fraction, would it be reasonable to put a default in the IDD?  Then you don't need to worry about it.  And for the curve, you could just set it to zero when blank, and then check for a zero value and issue the warning later after you know if there's waste heat (might need a flag to write the error only once).

File: doc/src/docs/InputOutputReference/01d-InputOutputReference.md [hunk 2]
```diff
@@ -5600,7 +5600,7 @@ Coil:Cooling:DX:TwoStageWithHumidityControlMode,
    CoilPerformance:DX:Cooling,                 !- Normal Mode Stage 1 Coil Performance Object Type
    DOAS Standard Perf 1,                             !- Normal Mode Stage 1 Coil Performance Name
    CoilPerformance:DX:Cooling,                 !- Normal Mode Stage 1+2 Coil Performance Object Type
-   DOAS Standard Perf 1+2,                         !- Normal Mode Stage 1+2 Coil Performance Name
+   DOAS Standard Perf 1+2,                  When the fuel type is not electricity and the parent object AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed does not require waste heat calculations, this field is ignored. If the field is blank, a warning will be issues and simulation continues. When the fuel type is not electricity and the parent object AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed requires waste heat calculations, if this field is left blank, the program assumes a constant value of 1 to make simulation continue and a warning will be issued.        !- Normal Mode Stage 1+2 Coil Performance Name
```
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01d-InputOutputReference.md:5603]: Something odd happened here.  But you can remove the dependency on fuel type here.

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 3]
```diff
@@ -1080,11 +1090,53 @@ namespace EnergyPlus {
...
+
+		CalcMultiSpeedDXCoilCooling( 1, 1, 1, 2, 1, 1 );
+
+		EXPECT_NEAR( 1303.4304, MSHPWasteHeat, 0.001 );
+
+		// clear
+		DXCoil.deallocate( );
+
+	}
+}
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/DXCoils.unit.cc]: Need a return at the end of this file.

Comment (lgu1234) [inline tst/EnergyPlus/unit/DXCoils.unit.cc]: @mjwitte I added a new line with a return.

File: doc/src/docs/InputOutputReference/01d-InputOutputReference.md (hunk 2 above)
Comment (lgu1234) [inline doc/src/docs/InputOutputReference/01d-InputOutputReference.md:5603]: @mjwitte Thanks. I cleaned it.

**Comment by lgu1234** [comment]
@mjwitte I don't know how to put a default in the IDD. I am going to remove checks of blank or invalid in GetDXCoils. Then when the heat recovery flag is available in InitDXCoil, I will issue a warning once to tell users the value of 1 is used when DXCoil( DXCoilNum ).MSWasteHeat == 0. What do you think?

**Comment by mjwitte** [comment]
in the IDD \default 0.2 for example.
In the getinput, you'll still need to check for a blank curve name to avoid calling the curvemanager and getting  "curve not found" error.  Just remove the warnings from there.

**Comment by lgu1234** [comment]
@mjwitte I am not clear how get default values from the schedule name field in the IDD. COuld you show me in detail?

**Comment by mjwitte** [comment]
I'm just suggesting that you change from this

```
  N14, \field Speed 1 Rated Waste Heat Fraction of Power Input
       \required-field
       \type real
       \units dimensionless
       \minimum> 0.0
       \maximum 1.0
       \note recoverable waste heat at full load and rated conditions
```

to this

```
  N14, \field Speed 1 Rated Waste Heat Fraction of Power Input
 DELETE-->      \required-field
       \type real
       \units dimensionless
ADD--> \default 0.2
       \minimum> 0.0
       \maximum 1.0
       \note recoverable waste heat at full load and rated conditions
```

If the field in blank the input processor will fill in the default long before the getinput reads it in.  For that field you don't need to check if blank ('cause that will still be true).  Just use the value in N14.  Same for N19 and for the two equivalent fields in coil:cooling:dx:multispeed.

**Comment by lgu1234** [comment]
@mjwitte I understand it now. If the schedule name filed is blank, I don't need to check.

**Comment by lgu1234** [comment]
@mjwitte I changed the files as mentioned and ran 4 test files to completion without extra warnings. I am going to make a commit. Please let me know what you think.

**Pull request merged**